[WIP] feat: introduce basic toast functionality

Izook opened the pull request:
> **⚠️WORK IN PROGRESS⚠️**

# Alaska Airlines Pull Request

# Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

- Initial implementation of auro-toast( and auro-toaster?). 
- This is addressing Issue [#189](https://github.com/AlaskaAirlines/auro/issues/189).

**Fixes:** # (issue, if applicable)

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

I have introduced a basic implementation of a toast notification based on [react-toast-notifications](https://jossmac.github.io/react-toast-notifications/).

## Type of change:

Please delete options that are not relevant.

- [x] New capability


## Checklist:

- [ ] My update follows the CONTRIBUTING guidelines of this project
- [ ] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/AlaskaAirlines/auro-toast?pullRequest=1) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/AlaskaAirlines/auro-toast?pullRequest=1) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/AlaskaAirlines/auro-toast?pullRequest=1) it.</sub>

Izook commented:
@blackfalcon I think I'm ready to get some design and auro feedback on this PR. 

Appearance wise here is what I have so far: 

![auro-toast](https://user-images.githubusercontent.com/23248722/100662541-e6830300-3309-11eb-803a-fa3fa082e68b.gif)

I have a question about the implementation though. Currently the way this is implemented this repo provides two components: `auro-toast` and `auro-toaster` allowing consumers to either use their own toaster for displaying toasts or use the prebuilt `auro-toaster`. We will probably want to move these into two different repos but I'm wondering that since the `auro-toaster` is a bit more complicated than other auro components would it make since to move it over to the component hangar and have it use `auro-toast` as a dependency or do we feel that we should keep it in auro?

blackfalcon commented on demo/index.html:
```diff
@@ -25,7 +26,37 @@ <h1>auro-toast</h1>
+          <h3>ACTIONS</h3>
+          <button onClick="clearToaster()">CLEAR TOASTER</button>
+
+          <div class="container" style="position: relative; width: 100%; height: 500px;">
```
If we are to use toaster as a custom element, this needs to be able to placed on the page anywhere and have absolut layering. It can't actually take up any physical space.

Izook replied:
Toaster removed. I think we can rethink that in another space.

blackfalcon reviewed (changes requested):
I am super impressed with this PR. There is a lot of really good stuff in here. Please don't take any of the comments as negative feedback. You did awesome and I really want to help you take this across the finish line.

Izook commented on src/auro-toast.js:
```diff
@@ -6,33 +6,91 @@
+   * @private function to set default properties within the constructor
+   * @returns {void}
+   */
+  setDefaultProperties() {
```
I've renamed the function to address this but the idea behind this function was to set the default value for all the _public_ properties for the auro-toast

Izook commented on src/auro-toast.js:
```diff
@@ -42,21 +100,133 @@ class AuroToast extends LitElement {
+      /* istanbul ignore next */
+      default:
+        this.svg = this.generateIconHtml(information.svg);
+        break;
```
Yup! Removed.

Izook commented on src/auro-toast.js:
```diff
@@ -42,21 +100,133 @@ class AuroToast extends LitElement {
+   */
+  updatePosition() {
+    const positions = [
+"left",
```
Adjusted!

Izook commented on src/auro-toaster.js:
```diff
@@ -0,0 +1,111 @@
+   */
+  updatePosition() {
+    const positions = [
+"top-left",
```
Adjusted!

Izook commented on src/style.scss:
```diff
@@ -10,26 +10,194 @@
 
-// Support for auroElement styles
-// @import "./node_modules/@alaskaairux/webcorestylesheets/dist/auroElement/auroElement";
+#toast {
```
I tried updating all the styles to match the auro css conventions. Let me know if there are any more spaces I can improve.

Izook commented on demo/index.html:
```diff
@@ -25,7 +26,37 @@ <h1>auro-toast</h1>
+          </script>
+
+          <h1>TYPE</h1>
+          <button onclick="addToast('success', 'top-left')">SUCCESS TOAST</button>
```
Thanks! I've since removed the auro-toaster from the demo and want to rethink that approach but have since re-implemented the swipe feature.

Izook commented on demo/index.html:
```diff
@@ -25,7 +26,37 @@ <h1>auro-toast</h1>
-          <auro-toast cssClass="testClass">Hello World!</auro-toast>
+          <script>
+            let position = "bottom-left";
+            let positions = ["bottom-left", "bottom-right", "top-left", "top-right"];
```
I've removed the positioning logic from the toaster to maintain a consistent bottom-right placement. As for the UI space stacking could you help me understand more how we would like these to be positioned or take up space?

Izook commented on demo/index.html:
```diff
@@ -25,7 +26,37 @@ <h1>auro-toast</h1>
+          <script>
+            let position = "bottom-left";
+            let positions = ["bottom-left", "bottom-right", "top-left", "top-right"];
+            function addToast(type) {
```
Are you saying we want to deliver a global `addToast` function along with npm?

Izook commented on demo/index.html:
```diff
@@ -25,7 +26,37 @@ <h1>auro-toast</h1>
+            function addToast(type) {
+              document.getElementById("toaster").addToast(document.createTextNode("This is a toast message!"), type, {});
+            }
+            function updatePosition(newPosition) {
```
Removed!

Izook commented:
Hello! I have finally gotten around to addressing some of your comments @blackfalcon! I've removed all of the `auro-toaster` code from the demo (still have to delete the code and tests) and worked on really simplifying the `auro-toast`. Here is what I have for the demo right now:

![ezgif-7-3a7611aef7cd](https://user-images.githubusercontent.com/23248722/108566601-89b93800-72bb-11eb-9ce2-6b8e8a9d8b8e.gif)

Let me know if this is more along the lines of what an auro-component should be delivering. I'm open to learn! I'd also love to work together with you to get the auro-toaster to a healthy place, and to rethink how or if we want to deliver an `auro-toaster` component if your available!

Izook commented:
Code and Stylesheets of current toast implementation on my account page: 
- https://dev.azure.com/itsals/E_Retain_Account/_git/Web-PreferenceCenter?path=%2Fpreference-center%2Fsrc%2Fstylesheets%2Fcomponents%2FCustomToast.module.scss
- https://dev.azure.com/itsals/E_Retain_Account/_git/Web-PreferenceCenter?path=%2Fpreference-center%2Fsrc%2Fcomponents%2Fshared%2FCustomToast.jsx

Izook commented:
New Auro Toast Design Spec: https://www.figma.com/file/5XP6Tv2F8M3mRqiVf2oZuV/Toasts?node-id=1%3A792
